WalletWasabi.Fluent/ViewModels/SearchBar/Settings/Setting.cs

```diff
@@ -1,43 +1,27 @@
-using System.Diagnostics.CodeAnalysis;
+using System.ComponentModel;
 using System.Linq.Expressions;
+using System.Reactive.Disposables;
 using System.Reactive.Linq;
-using System.Windows.Input;
 using ReactiveUI;
-using WalletWasabi.Fluent.Extensions;
 
 namespace WalletWasabi.Fluent.ViewModels.SearchBar.Settings;
 
-public partial class Setting<TTarget, TProperty> : ReactiveObject
+public class Setting<TOwner, TProperty> : ReactiveObject, IDisposable where TOwner : class, INotifyPropertyChanged
 {
-	[AutoNotify] private TProperty? _value;
+	private readonly CompositeDisposable _disposable = new();
+	private TProperty _value = default!;
 
-	public Setting([DisallowNull] TTarget target, Expression<Func<TTarget, TProperty>> selector)
+	public Setting(TOwner owner, Expression<Func<TOwner, TProperty?>> propertySelector)
 	{
-		if (target == null)
-		{
-			throw new ArgumentNullException(nameof(target));
-		}
-
-		if (selector == null)
-		{
-			throw new ArgumentNullException(nameof(selector));
-		}
-
-		if (PropertyHelper<TTarget>.GetProperty(selector) is not { } pr)
-		{
-			throw new InvalidOperationException($"The expression {selector} is not a valid property selector");
-		}
-
-		Value = (TProperty?)pr.GetValue(target);
-
-		SetValueCommand = ReactiveCommand.Create(() => pr.SetValue(target, Value));
+		owner.WhenAnyValue(propertySelector).BindTo(this, value => value.Value).DisposeWith(_disposable);
+		this.WhenAnyValue(x => x.Value).Skip(1).BindTo(owner, propertySelector).DisposeWith(_disposable);
+	}
 
-		this.WhenAnyValue(x => x.Value)
-			.ObserveOn(RxApp.MainThreadScheduler)
-			.Skip(1)
-			.ToSignal()
-			.InvokeCommand(SetValueCommand);
+	public TProperty Value
+	{
+		get => _value;
+		set => this.RaiseAndSetIfChanged(ref _value, value);
 	}
 
-	public ICommand SetValueCommand { get; }
+	public void Dispose() => _disposable.Dispose();
```

Comment:
Where is this called? Shouldn't the `SettingsSearchSource` L26 contain `DisposeMany()`?

Reply:
Yes :) That should be fixed once https://github.com/zkSNACKs/WalletWasabi/pull/12625/files is merged.